pyasic/miners/antminer/bmminer/X19/X19.py

```diff
@@ -25,131 +25,74 @@
 
 
 class BMMinerX19(BMMiner):
-    def __init__(self, ip: str, api_ver: str = "1.0.0") -> None:
+    def __init__(self, ip: str, api_ver: str = "0.0.0") -> None:
         super().__init__(ip, api_ver=api_ver)
         self.ip = ip
         self.uname = "root"
         self.pwd = PyasicSettings().global_x19_password
 
-    async def check_light(self) -> Union[bool, None]:
-        if self.light:
-            return self.light
-        url = f"http://{self.ip}/cgi-bin/get_blink_status.cgi"
+    async def send_web_command(
+        self, command: str, params: dict = None
+    ) -> Optional[dict]:
+        url = f"http://{self.ip}/cgi-bin/{command}.cgi"
         auth = httpx.DigestAuth(self.uname, self.pwd)
-        async with httpx.AsyncClient() as client:
-            data = await client.get(url, auth=auth)
-        if data.status_code == 200:
-            data = data.json()
-            light = data["blink"]
-            self.light = light
-            return light
-        return None
+        try:
+            async with httpx.AsyncClient() as client:
+                if params:
+                    data = await client.post(url, data=params, auth=auth)
+                else:
+                    data = await client.get(url, auth=auth)
+        except httpx.HTTPError:
+            pass
+        else:
+            if data.status_code == 200:
+                try:
+                    return data.json()
+                except json.decoder.JSONDecodeError:
+                    pass
 
     async def get_config(self) -> MinerConfig:
-        url = f"http://{self.ip}/cgi-bin/get_miner_conf.cgi"
-        auth = httpx.DigestAuth(self.uname, self.pwd)
-        async with httpx.AsyncClient() as client:
-            data = await client.get(url, auth=auth)
-        if data.status_code == 200:
-            data = data.json()
+        data = await self.send_web_command("get_miner_conf")
+        if data:
             self.config = MinerConfig().from_raw(data)
         return self.config
 
     async def send_config(self, config: MinerConfig, user_suffix: str = None) -> None:
-        url = f"http://{self.ip}/cgi-bin/set_miner_conf.cgi"
-        auth = httpx.DigestAuth(self.uname, self.pwd)
         conf = config.as_x19(user_suffix=user_suffix)
+        await self.send_web_command(
+            "set_miner_conf", params=conf  # noqa: ignore conf being a str
+        )
 
-        try:
-            async with httpx.AsyncClient() as client:
-                await client.post(url, data=conf, auth=auth)  # noqa - ignore conf being a str
-        except httpx.ReadTimeout:
-            pass
         for i in range(7):
             data = await self.get_config()
             if data.as_x19() == conf:
                 break
             await asyncio.sleep(1)
 
-    async def get_hostname(self) -> Union[str, None]:
```

Comment:
Where did get_hostname go? Realizing that I'm looking through this much later and that doing it in person would have been more efficient

Reply:
I believe it got moved up a level maybe?  I can find an S19 to test on and call `get_hostname` on it tomorrow, unless you want to do it tonight?

Reply:
https://github.com/UpstreamData/pyasic/blob/dev-fixed/pyasic/miners/antminer/bmminer/X19/X19.py

Line 108.